index.js

```diff
@@ -217,6 +218,9 @@ module.exports = function (filename, opts) {
   function writeBlock(blockIndex) {
     const { block, fileOffset } = blocksToBeWritten[blockIndex]
     delete blocksToBeWritten[blockIndex]
+    let wd = waitingDrain[blockIndex] || []
+    const drain = wd.slice(0)
```

Comment:
Shouldn't these be inside the `raf.write` callback?

(Optional, can merge without this: a bit more readable code would be `const drainsBefore = (waitingDrain[blockIndex] || []).slice(0)`, avoiding a temp `wd` as a `let`, and `drainsBefore` named consistently like `drainsAfter` is)

Reply:
I'm pretty sure it needs to be here, so we know which ones were ready before we do the async call. Good idea with the names, will fix that.

Reply:
Oh, about the first sentence: are they outside the `raf.write` callback because they're supposed to reflect exactly this block? I thought it was like a concurrency protection thing (hence the clone via `.slice(0)`)